CONTRIBUTING.md

```diff
@@ -0,0 +1,291 @@
+  frontend (JavaScript), mobile (React Native), desktop (Electron),
+  documentation (English) or visual design (JavaScript + CSS) issue, and a
+  bit about your programming experience.
+
+We also welcome suggestions of features that you feel would be valuable or
+changes that you feel would make Zulip a better open source project. If you
+have a new feature you'd like to add, we recommend you start by posting in
+[#new members](https://chat.zulip.org/#narrow/stream/new.20members) with the
+feature idea and the problem that you're hoping to solve.
+
+Other notes:
+* For a first pull request, it's better to aim for a smaller contribution
+  than a bigger one. Many first contributions have fewer than 10 lines of
+  changes (not counting tests).
+* The full list of issues looking for a contributor can be found with the
+  [help wanted](https://github.com/zulip/zulip/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
+  label.
+* For most new contributors, there's a lot to learn while making your first
+  pull request. It's OK if it takes you a while; that's normal! You'll be
+  able to work a lot faster as you build experience.
+
+### Working on an issue
+
+To work on an issue, claim it by adding a comment with `@zulipbot claim` to
+the issue thread. [Zulipbot](https://github.com/zulip/zulipbot) is a GitHub
+workflow bot; it will assign you to the issue and label the issue as "in
+progress". Some additional notes:
+
+* You're encouraged to ask questions on how to best implement or debug your
+  changes -- the Zulip maintainers are excited to answer questions to help
+  you stay unblocked and working efficiently. You can ask questions on
+  chat.zulip.org, or on the GitHub issue or pull request.
+* We encourage early pull requests for work in progress. Prefix the title of
+  work in progress pull requests with `[WIP]`, and remove the prefix when
+  you think it might be mergeable and want it to be reviewed.
+* After updating a PR, add a comment to the GitHub thread mentioning that it
+  is ready for another review. GitHub only notifies maintainers of the
+  changes when you post a comment, so if you don't, your PR will likely be
+  neglected by accident!
+
+### And beyond
+
+A great place to look for a second issue is to look for issues with the same
+`area:` label as the last issue you resolved. You'll be able to reuse the
+work you did learning how that part of the codebase works. Also, the path to
+becoming a core developer often involves taking ownership of one of these area
+labels.
+
+## What makes a great Zulip contributor?
+
+Zulip runs a lot of internship programs, so we have a lot of experience with
+new contributors. In our experience, these are the best predictors of success:
+
+* Posting good questions. This generally means explaining your current
+  understanding, saying what you've done or tried so far, and including
+  tracebacks or other error messages if appropriate.
```

Comment:
Maybe include a recommendation to use code blocks rather than screenshots?  Probably just a thing to add a link to a doc on screenshots and screencasts; feel free to put this in a follow-up issue.